server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanCommunicationDetailsCacheTest.java

```diff
@@ -61,4 +61,46 @@ public void testSingleConsumerFound() {
         assertEquals(cd, cdc.get(null, "id1"));
     }
 
+    @Test
+    public void testGetByIdMultiple() throws CacheException {
+        InfinispanCommunicationDetailsCache cdc = createCache();
+
+        CommunicationDetails cdMultipleConsumer1 = new CommunicationDetails();
+        cdMultipleConsumer1.setId("id1");
+        cdMultipleConsumer1.setTargetFragmentId("fragmentId1");
+        cdMultipleConsumer1.setMultiConsumer(true);
+
+        CommunicationDetails cdMultipleConsumer2 = new CommunicationDetails();
+        cdMultipleConsumer2.setId("id1");
+        cdMultipleConsumer2.setTargetFragmentId("fragmentId2");
+        cdMultipleConsumer2.setMultiConsumer(true);
+
+        CommunicationDetails cd3 = new CommunicationDetails();
+        cd3.setId("id2");
+
+        cdc.store(null, Arrays.asList(cdMultipleConsumer1, cdMultipleConsumer2, cd3));
+
+        List<CommunicationDetails> cdFromCache = cdc.getById(null, "id1");
+
+        Assert.assertEquals(2, cdFromCache.size());
+        Assert.assertEquals(new HashSet<>(Arrays.asList(cdMultipleConsumer1, cdMultipleConsumer2)),
```

Comment:
Set from mathematical point of view is unordered collection so order does not matter:
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/util/AbstractSet.java#85
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/util/AbstractCollection.java#AbstractCollection.containsAll%28java.util.Collection%29

And here I'm assuming that the orrder does not matter, I  don't think that infinispan stores values in some order (e.g. order of the insertion). It internally uses some hash table. I doubt that is is some red black/B+ trees